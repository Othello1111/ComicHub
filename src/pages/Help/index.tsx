import React, { useEffect } from 'react';
import Store from '../../store';

type Props = {}
const Help: React.FunctionComponent<Props> = () => {
  const { toggleLoading } = Store.useContainer();
  useEffect(() => {
    toggleLoading();
  }, []);
  return <div>Help</div>;
};
export default Help;
