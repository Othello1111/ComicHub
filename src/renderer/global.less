@import '~antd/lib/style/themes/default.less';

body {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #000;
  line-height: 1.5;
  background-color: #f2f4f5;
}
