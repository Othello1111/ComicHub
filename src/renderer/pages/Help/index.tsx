import React from 'react';

const Help: React.FunctionComponent = () => <div>Help</div>;
export default Help;
