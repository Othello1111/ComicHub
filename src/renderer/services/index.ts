import request from '../utils/request';

interface PostData {
    url?: string;

    [index: string]: string;
}

export async function getMenuList() {
    return request('/v1/menu');
}

export async function postItem(params: PostData = {}) {
    const { url, ...rest } = params;
    if (!url) {
        return;
    }
    const realUrl = `/v1/${url}`;
    return request(realUrl, {
        method: 'POST',
        body: rest,
    });
}
