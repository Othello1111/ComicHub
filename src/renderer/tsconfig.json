{
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        "importHelpers": true,
        "jsx": "preserve",
        "esModuleInterop": true,
        "sourceMap": true,
        "strict": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "baseUrl": ".",
        "paths": {
            "@/*": ["./*"]
        },
        "allowSyntheticDefaultImports": true
    }
}
