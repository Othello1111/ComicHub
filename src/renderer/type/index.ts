export interface SharedState {
    currentUrl: string;
    currentType: string;
}
export interface MenuItem {
    value: string;
    enabled: boolean;
    base: string;
    name: string;
}
export interface IFormData {
    url: string;
    name: string;
}

export interface IOptionData {
    value: string | number;
    name: string;
    enabled: boolean;
}
