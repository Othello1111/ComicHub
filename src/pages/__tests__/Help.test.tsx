import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import Help from '../Help';
import Store from '../../store';
describe('Page: Help', () => {
  it('Render correctly', () => {
    const wrapper: ReactTestRenderer = renderer.create(<Store.Provider><Help/></Store.Provider>);
    expect(wrapper.root.children.length).toBeGreaterThan(1);
  });
  it('Page: Help Snapshots', () => {
    const tree = renderer.create(<Store.Provider><Help/></Store.Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
