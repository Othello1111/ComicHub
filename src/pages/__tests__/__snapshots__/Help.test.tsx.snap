// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Help Page: Help Snapshots 1`] = `
Array [
  " ",
  <div>
    Help
  </div>,
  " ",
]
`;
